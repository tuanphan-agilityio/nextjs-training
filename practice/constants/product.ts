const SIZES = ['S', 'M', 'L', 'XL'];

export { SIZES };
