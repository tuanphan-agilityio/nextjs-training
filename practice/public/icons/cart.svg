<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="Frame 5">
<path id="Rectangle 5" d="M6.12403 9.03885C6.2568 7.97675 7.42732 7.39149 8.35666 7.92254L10.7597 9.29568C11.5282 9.73487 12.4718 9.73487 13.2403 9.29568L15.6433 7.92254C16.5727 7.39149 17.7432 7.97675 17.876 9.03885L19.2229 19.8139C19.3348 20.7092 18.6367 21.5 17.7344 21.5H6.26556C5.36332 21.5 4.66524 20.7092 4.77715 19.8139L6.12403 9.03885Z" stroke="#0D1222"/>
<path id="Vector" d="M15 6V4.01042C15 2.89583 14.0076 2 12.7985 2H11.2015C9.98099 2 9 2.90625 9 4.01042V6" stroke="#0D1222" stroke-width="1.2" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
<circle id="Ellipse 3" cx="9" cy="12" r="0.8" stroke="#0D1222" stroke-width="0.4"/>
<circle id="Ellipse 4" cx="15" cy="12" r="0.8" stroke="#0D1222" stroke-width="0.4"/>
</g>
</svg>
