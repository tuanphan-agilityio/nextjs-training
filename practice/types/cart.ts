interface CartItem {
  quantity: number;
  productId: number;
}

export type { CartItem };
