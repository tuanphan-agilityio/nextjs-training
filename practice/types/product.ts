interface Product {
  id: number;
  price: number;
  name: string;
  thumbnail: string;
  status: string;
  imgHrefs: string[];
  description: string;
  stock: number;
}

export type { Product };
