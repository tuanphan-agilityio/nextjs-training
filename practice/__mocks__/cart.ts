import { CartProductItem, CartStorage } from '@/types/cart';

const CART_STORAGES: CartStorage[] = [
  { productId: 1, quantity: 2 },
  { productId: 2, quantity: 2 },
  { productId: 3, quantity: 2 },
];

const CART_ITEMS: CartProductItem[] = CART_STORAGES.map(
  ({ productId, quantity }, index) => ({
    productId,
    quantity,
    name: `Product ${index}`,
    thumbnail: `Thumbnail ${index}`,
    price: 30,
    stock: 1305,
  }),
);

export { CART_STORAGES, CART_ITEMS };
