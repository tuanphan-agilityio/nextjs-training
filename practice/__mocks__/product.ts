import { Product } from '@/types/product';

const MOCK_PRODUCT: Product = {
  id: 1,
  name: 'Smart T-shirt',
  price: 120,
  status: 'Best quality',
  thumbnail: '/images/product-2.jpg',
  imgHrefs: ['/images/product-1.jpg', '/images/product-3.jpg'],
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. In eget gravida leo, nec iaculis diam. Nam bibendum mi sed sem finibus ullamcorper.',
  stock: 1305,
};

const MOCK_PRODUCTS: Product[] = Array(3)
  .fill(MOCK_PRODUCT)
  .map((product, index) => ({ ...product, id: index + 1 }));

const MOCK_DEFAULT_PARAMS = {
  _page: 1,
  _limit: 3,
};

export { MOCK_PRODUCT, MOCK_PRODUCTS, MOCK_DEFAULT_PARAMS };
