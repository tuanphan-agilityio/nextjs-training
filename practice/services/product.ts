import { Product } from '@/types/product';
import axiosApp from './axiosApp';
import { ENDPOINTS } from '@/constants/endpoints';
import { Pagination } from '@/types/common';

const getProducts = async (params?: Pagination): Promise<Product[]> => {
  const response: Product[] = await axiosApp.get(ENDPOINTS.PRODUCTS, {
    params,
  });

  return response;
};

const getProduct = async (id: string): Promise<Product> => {
  const response: Product = await axiosApp.get(`${ENDPOINTS.PRODUCTS}/${id}`);

  return response;
};

export { getProducts, getProduct };
