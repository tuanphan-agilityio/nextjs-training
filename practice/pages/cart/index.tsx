import { FC, useCallback, useEffect, useMemo, useState } from 'react';
import Head from 'next/head';

import SubHeader from '@/components/SubHeader';
import Button from '@/components/Button';
import Input from '@/components/Input';
import Modal from '@/components/Modal';
import CartItem from '@/components/CartItem';

import { LocalStorage } from '@/utils/storage';
import { transformData } from '@/utils/cart/transformData';

import { STORAGE_KEYS } from '@/constants/storage';

import { CartProductItem, CartStorage } from '@/types/cart';

import { getProductsByIds } from '@/services/product';

import { updateLocalStorageCart } from '@/utils/cart/updateLocalStorageCart';

const CartPage: FC = () => {
  const [cart, setCart] = useState<CartProductItem[]>([]);

  const [productIdConsiderDelete, setProductIdConsiderDelete] = useState<
    number | null
  >(null);

  const breadCrumbItems = useMemo(() => [{ label: 'Cart' }], []);

  const totalPrice = useMemo(() => {
    return cart.reduce(
      (accumulator, currentValue) =>
        accumulator + currentValue.price * currentValue.quantity,
      0,
    );
  }, [cart]);

  const handleQuantityChange = useCallback(
    (value: number, id: number) => {
      const updatedCart = cart.map((item) => {
        if (item.productId === id) {
          return { ...item, quantity: value };
        }
        return item;
      });

      updateLocalStorageCart(updatedCart);
      setCart(updatedCart);
    },
    [setCart, cart],
  );

  const handleDeleteCartItem = useCallback((id: number) => {
    setProductIdConsiderDelete(id);
  }, []);

  const handleOnCancelPopup = useCallback(() => {
    setProductIdConsiderDelete(null);
  }, []);

  const handleOnOkPopup = useCallback(() => {
    if (productIdConsiderDelete) {
      const updatedCart = cart.filter(
        (item) => item.productId !== productIdConsiderDelete,
      );

      updateLocalStorageCart(updatedCart);
      setCart(updatedCart);
      setProductIdConsiderDelete(null);
    }
  }, [productIdConsiderDelete, cart]);

  const renderCheckout = useMemo(() => {
    return (
      <div className='py-10 px-6 w-1/3 bg-quinary text-md'>
        <p className='mb-2 font-secondary-bold'>Coupon Code</p>
        <Input placeholder='Enter your coupon code' />

        <div className='text-right'>
          <Button className='mt-4'>Check</Button>
        </div>

        <div className='mt-9 pb-6 flex justify-between border-b-2 border-dashed border-tertiary'>
          <div>
            <p className='mb-2 font-secondary-bold'>Subtotal</p>
            <p className='text-tertiary'>{cart.length} Product</p>
          </div>
          <p className='font-secondary-regular'>$ {totalPrice} USD</p>
        </div>

        <div className='my-6 flex justify-between'>
          <p className='mb-2 font-secondary-bold'>Total</p>
          <p className='font-secondary-regular'>$ {totalPrice} USD</p>
        </div>

        <div className='text-right'>
          <Button>Checkout</Button>
        </div>
      </div>
    );
  }, [cart.length, totalPrice]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const cartStorages =
          (LocalStorage.load(STORAGE_KEYS.CART) as CartStorage[]) ?? [];

        const productIds = cartStorages.map((cartStorage) =>
          String(cartStorage.productId),
        );

        const quantities = cartStorages.map(
          (cartStorage) => cartStorage.quantity,
        );
        const data = await getProductsByIds(productIds);

        setCart(transformData(data, quantities));
        // eslint-disable-next-line no-empty
      } catch (error) {}
    };

    fetchData();
  }, []);

  return (
    <>
      <main>
        <Head>
          <title>Cart | Checkout</title>
        </Head>
        <SubHeader
          title='Cart'
          breadcrumbItems={breadCrumbItems}
          description='Lorem ipsum dolor sit amet, consectetur adipiscing elit. In eget gravida leo, nec iaculis diam. Nam bibendum mi sed sem finibus ullamcorper.'
        />
        <div className='container py-20'>
          <h2 className='mb-10 text-3xl font-secondary-bold leading-6'>
            Cart Product
          </h2>

          <div className='flex gap-10'>
            <div className='flex flex-col gap-6 w-2/3 font-secondary-bold text-secondary text-xl'>
              {cart.map(
                ({ productId, name, price, stock, quantity, thumbnail }) => (
                  <CartItem
                    key={productId}
                    productId={productId}
                    name={name}
                    price={price}
                    stock={stock}
                    quantity={quantity}
                    imgHref={thumbnail}
                    onQuantityChange={handleQuantityChange}
                    onDeleteCartItem={handleDeleteCartItem}
                  />
                ),
              )}
            </div>
            {renderCheckout}
          </div>
        </div>
      </main>
      <Modal
        isOpen={!!productIdConsiderDelete}
        onCancel={handleOnCancelPopup}
        onOk={handleOnOkPopup}
        isConfirmModal
      >
        <p className='text-md font-primary-bold'>
          Are you sure you want to delete this product?
        </p>
      </Modal>
    </>
  );
};

export default CartPage;
