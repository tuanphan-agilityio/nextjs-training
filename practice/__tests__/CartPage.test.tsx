import { render, screen, fireEvent } from '@testing-library/react';
import CartPage from '@/pages/cart/index';
import { act } from 'react-dom/test-utils';
import { MOCK_PRODUCT } from '@/__mocks__/product';
import { CART_STORAGES } from '@/__mocks__/cart';

jest.mock('@/services/product', () => ({
  getProduct: jest.fn().mockImplementation((productId: string) => ({
    ...MOCK_PRODUCT,
    id: productId,
  })),
}));

describe('CartPage', () => {
  afterAll(() => {
    window.localStorage.clear();
  });

  it('Cart Page render correctly products', async () => {
    window.localStorage.setItem('shop_cart', JSON.stringify(CART_STORAGES));
    const { container } = render(<CartPage />);

    expect(await screen.findAllByText(/\$ 120 usd/i)).toBeInTheDocument;

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should delete a product from the cart', async () => {
    window.localStorage.setItem('shop_cart', JSON.stringify(CART_STORAGES));

    render(<CartPage />);

    await screen.findAllByText('Cart Product');

    expect(screen.getAllByText('Smart T-shirt')[0]).toBeInTheDocument();

    const deleteButton = screen.getAllByTestId('delete-button')[0];
    act(() => {
      fireEvent.click(deleteButton);
    });

    await screen.findByText('Are you sure you want to delete this product?');

    const confirmDeleteButton = screen.getByText('Confirm');
    act(() => {
      fireEvent.click(confirmDeleteButton);
    });

    expect(await screen.findAllByText('Smart T-shirt')).toHaveLength(
      CART_STORAGES.length - 1,
    );
  });
});
