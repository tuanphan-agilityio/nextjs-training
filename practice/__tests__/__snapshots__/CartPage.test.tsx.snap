// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPage Cart Page render correctly products 1`] = `
<main>
  <section
    class="pt-10 pb-20 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary md:pt-8 md:pb-14 sm:pt-4 sm:pb-4"
  >
    <div
      class="container"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="flex w-fit text-sm text-primary"
        >
          <li
            class="font-primary-bold"
          >
            Cart
          </li>
        </ol>
      </nav>
      <h1
        class="mt-10 text-3xl font-secondary-bold md:mt-6 sm:mt-4"
      >
        Cart
      </h1>
      <p
        class="mt-6 w-[508px] font-md text-justify leading-2 sm:w-full"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. In eget gravida leo, nec iaculis diam. Nam bibendum mi sed sem finibus ullamcorper.
      </p>
    </div>
  </section>
  <div
    class="container py-20 xl:py-16 lg:py-12 md:my-6 sm:py-4"
  >
    <h2
      class="mb-10 text-3xl font-secondary-bold "
    >
      Cart Product
    </h2>
    <div
      class="flex gap-10 lg:gap-4 lg:flex-col"
    >
      <div
        class="flex flex-col gap-6 w-2/3 font-secondary-bold text-secondary text-xl lg:w-full sm:text-sm"
      >
        <ul
          class="flex gap-4 p-10 bg-quinary xl:p-4 sm:flex-wrap sm:justify-center sm:p-2"
        >
          <li
            class="w-1/5"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iODAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                alt="Product"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </li>
          <li
            class="w-2/5"
          >
            <h3
              class="mb-2 truncate"
            >
              Smart T-shirt
            </h3>
            <p
              class="cart-item-subtext"
            >
              Color : White
            </p>
          </li>
          <li
            class="w-1/5"
          >
            <p
              class="truncate mb-2"
            >
              $ 
              120
               USD
            </p>
            <p
              class="cart-item-subtext"
            >
              Price
            </p>
          </li>
          <li
            class="w-1/5 min-w-[142px]"
          >
            <div
              class="flex justify-between"
            >
              <div>
                <p
                  class="truncate mb-2"
                >
                  1305
                </p>
                <p
                  class="cart-item-subtext whitespace-nowrap"
                >
                  In Stock
                </p>
              </div>
              <button
                class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 text-transparent font-primary-bold bg-clip-text bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary w-fit"
                data-testid="delete-button"
              >
                <svg
                  fill="none"
                  height="24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.53 21.13H7.06c-1.1 0-2-.9-2-2V5.61h13.47v13.52c0 1.1-.9 2-2 2ZM2.6 5.61h18.39M14.56 5.61V3.6c0-.55-.45-1-1-1h-3.54c-.55 0-1 .45-1 1v2.01M9.68 8.84v7.96M13.9 8.84v7.96"
                    stroke="#0D1222"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    stroke-width="1.2"
                  />
                </svg>
              </button>
            </div>
            <div
              class="mt-11 text-md md:mt-6 sm:mt-2 sm:text-sm"
            >
              <div
                class="flex items-center p-1 w-[142px] font-secondary-bold border border-secondary bg-primary sm:h-8"
              >
                <button
                  class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-none text-secondary font-primary-bold hover:bg-quinary sm:h-7"
                >
                  -
                </button>
                
                <div
                  class="flex px-4 py-2 w-full bg-primary sm:h-7"
                >
                  <input
                    aria-label="counter number"
                    class="block w-full border-none focus:outline-none text-center"
                    value="2"
                  />
                </div>
                <button
                  class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-none text-secondary font-primary-bold hover:bg-quinary sm:h-7"
                >
                  +
                </button>
              </div>
            </div>
          </li>
        </ul>
        <ul
          class="flex gap-4 p-10 bg-quinary xl:p-4 sm:flex-wrap sm:justify-center sm:p-2"
        >
          <li
            class="w-1/5"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iODAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                alt="Product"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </li>
          <li
            class="w-2/5"
          >
            <h3
              class="mb-2 truncate"
            >
              Smart T-shirt
            </h3>
            <p
              class="cart-item-subtext"
            >
              Color : White
            </p>
          </li>
          <li
            class="w-1/5"
          >
            <p
              class="truncate mb-2"
            >
              $ 
              120
               USD
            </p>
            <p
              class="cart-item-subtext"
            >
              Price
            </p>
          </li>
          <li
            class="w-1/5 min-w-[142px]"
          >
            <div
              class="flex justify-between"
            >
              <div>
                <p
                  class="truncate mb-2"
                >
                  1305
                </p>
                <p
                  class="cart-item-subtext whitespace-nowrap"
                >
                  In Stock
                </p>
              </div>
              <button
                class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 text-transparent font-primary-bold bg-clip-text bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary w-fit"
                data-testid="delete-button"
              >
                <svg
                  fill="none"
                  height="24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.53 21.13H7.06c-1.1 0-2-.9-2-2V5.61h13.47v13.52c0 1.1-.9 2-2 2ZM2.6 5.61h18.39M14.56 5.61V3.6c0-.55-.45-1-1-1h-3.54c-.55 0-1 .45-1 1v2.01M9.68 8.84v7.96M13.9 8.84v7.96"
                    stroke="#0D1222"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    stroke-width="1.2"
                  />
                </svg>
              </button>
            </div>
            <div
              class="mt-11 text-md md:mt-6 sm:mt-2 sm:text-sm"
            >
              <div
                class="flex items-center p-1 w-[142px] font-secondary-bold border border-secondary bg-primary sm:h-8"
              >
                <button
                  class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-none text-secondary font-primary-bold hover:bg-quinary sm:h-7"
                >
                  -
                </button>
                
                <div
                  class="flex px-4 py-2 w-full bg-primary sm:h-7"
                >
                  <input
                    aria-label="counter number"
                    class="block w-full border-none focus:outline-none text-center"
                    value="2"
                  />
                </div>
                <button
                  class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-none text-secondary font-primary-bold hover:bg-quinary sm:h-7"
                >
                  +
                </button>
              </div>
            </div>
          </li>
        </ul>
        <ul
          class="flex gap-4 p-10 bg-quinary xl:p-4 sm:flex-wrap sm:justify-center sm:p-2"
        >
          <li
            class="w-1/5"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iODAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                alt="Product"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </li>
          <li
            class="w-2/5"
          >
            <h3
              class="mb-2 truncate"
            >
              Smart T-shirt
            </h3>
            <p
              class="cart-item-subtext"
            >
              Color : White
            </p>
          </li>
          <li
            class="w-1/5"
          >
            <p
              class="truncate mb-2"
            >
              $ 
              120
               USD
            </p>
            <p
              class="cart-item-subtext"
            >
              Price
            </p>
          </li>
          <li
            class="w-1/5 min-w-[142px]"
          >
            <div
              class="flex justify-between"
            >
              <div>
                <p
                  class="truncate mb-2"
                >
                  1305
                </p>
                <p
                  class="cart-item-subtext whitespace-nowrap"
                >
                  In Stock
                </p>
              </div>
              <button
                class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 text-transparent font-primary-bold bg-clip-text bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary w-fit"
                data-testid="delete-button"
              >
                <svg
                  fill="none"
                  height="24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16.53 21.13H7.06c-1.1 0-2-.9-2-2V5.61h13.47v13.52c0 1.1-.9 2-2 2ZM2.6 5.61h18.39M14.56 5.61V3.6c0-.55-.45-1-1-1h-3.54c-.55 0-1 .45-1 1v2.01M9.68 8.84v7.96M13.9 8.84v7.96"
                    stroke="#0D1222"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-miterlimit="10"
                    stroke-width="1.2"
                  />
                </svg>
              </button>
            </div>
            <div
              class="mt-11 text-md md:mt-6 sm:mt-2 sm:text-sm"
            >
              <div
                class="flex items-center p-1 w-[142px] font-secondary-bold border border-secondary bg-primary sm:h-8"
              >
                <button
                  class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-none text-secondary font-primary-bold hover:bg-quinary sm:h-7"
                >
                  -
                </button>
                
                <div
                  class="flex px-4 py-2 w-full bg-primary sm:h-7"
                >
                  <input
                    aria-label="counter number"
                    class="block w-full border-none focus:outline-none text-center"
                    value="2"
                  />
                </div>
                <button
                  class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-none text-secondary font-primary-bold hover:bg-quinary sm:h-7"
                >
                  +
                </button>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="py-10 px-6 w-1/3 bg-quinary text-md lg:w-full"
      >
        <p
          class="mb-2 font-secondary-bold"
        >
          Coupon Code
        </p>
        
        <div
          class="flex px-4 py-2 w-full bg-primary"
        >
          <input
            class="block w-full border-none focus:outline-none text-left"
            placeholder="Enter your coupon code"
          />
        </div>
        <div
          class="text-right sm:text-center"
        >
          <button
            class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary mt-4"
          >
            Check
          </button>
        </div>
        <div
          class="mt-9 pb-6 flex justify-between border-b-2 border-dashed border-tertiary"
        >
          <div>
            <p
              class="mb-2 font-secondary-bold"
            >
              Subtotal
            </p>
            <p
              class="text-tertiary"
            >
              3
               Product
            </p>
          </div>
          <p
            class="font-secondary-regular"
          >
            $ 
            720
             USD
          </p>
        </div>
        <div
          class="my-6 flex justify-between"
        >
          <p
            class="mb-2 font-secondary-bold"
          >
            Total
          </p>
          <p
            class="font-secondary-regular"
          >
            $ 
            720
             USD
          </p>
        </div>
        <div
          class="text-right sm:text-center"
        >
          <button
            class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary"
          >
            Checkout
          </button>
        </div>
      </div>
    </div>
  </div>
</main>
`;
